docs/docs/core/content-manager/hooks/use-drag-and-drop.mdx

```diff
@@ -0,0 +1,212 @@
+      ) : (
+        <Flex ref={composedRefs} data-handler-id={handlerId}>
+          <IconButton
+            forwardedAs="div"
+            role="button"
+            tabIndex={0}
+            aria-label="Drag"
+            noBorder
+            onKeyDown={handleKeyDown}
+          >
+            <Drag />
+          </IconButton>
+          {'My item'}
+        </Flex>
+      )}
+    </Box>
+  );
+};
+```
+
+## Typescript
+
+```ts
+import { Identifier } from 'dnd-core';
+import { ConnectDropTarget, ConnectDragSource, ConnectDragPreview } from 'react-dnd';
+
+interface UseDragAndDropOptions {
+  index: number;
+  onMoveItem: (newIndex: number, currentIndex: number) => void;
+  item?: object;
+  type?: string;
+  onCancel?: (index: number) => void;
+  onDropItem?: (index: number) => void;
+  onEnd?: () => void;
+  onGrabItem?: (index: number) => void;
+  onStart?: () => void;
+}
+
+type UseDragAndDropReturn = [
+  props: {
+    handlerId: Identifier;
+    isDragging: boolean;
+    handleKeyDown: (event: KeyboardEvent<HTMLButtonElement>) => void;
+  },
+  objectRef: React.RefObject<HTMLElement>,
+  dropRef: ConnectDropTarget,
+  dragRef: ConnectDragSource,
+  dragPreviewRef: ConnectDragPreview
+];
+
+type UseDragAndDrop = (active: boolean, options: UseDragAndDropOptions) => UseDragAndDropReturn;
+```
+
+## Accessibility
+
+Its advised to implement a [live text region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) in the
+parent component holding your individual dnd children. This should be done to inform the user of the current state of the drag and drop.
+To implement this, you need to pass the `onDropItem`, `onGrabItem` and `onCancel` callbacks to the `useDragAndDrop` hook which are fired
+only with the purpose of updating the live region, hence why they're optional. You would also update the live region as part of your
+`onMoveItem` callback. There are generic messages that can be used in the `intl` provider, an example of using this may look like:
+
+```js
+setLiveText(
+  formatMessage(
+    {
+      id: getTrad('dnd.drop-item'),
+      defaultMessage: `{item}, dropped. Final position in list: {position}.`,
+    },
+    {
+      item: 'my item',
+      position: 1,
+    }
+  )
+);
+```
+
+## Further Reading
+
+- [react-dnd docs](https://react-dnd.github.io/react-dnd/docs/overview)
+- [useDrag API](https://react-dnd.github.io/react-dnd/docs/api/use-drag)
+- [useDrop API](https://react-dnd.github.io/react-dnd/docs/api/use-drop)
+- [useDragLayer API](https://react-dnd.github.io/react-dnd/docs/api/use-drag-layer)
+
+## Troubleshooting
+
+### Firefox quirks
+
+You might notice in the [basic usage](#basic-usage) section this piece of code:
+
+```jsx
+<IconButton
+  forwardedAs="div"
+  role="button"
+  tabIndex={0}
+  aria-label="Drag"
+  noBorder
+  onKeyDown={handleKeyDown}
+>
+  <Drag />
+</IconButton>
+```
+
+In `firefox` the drag handler will not work if you click and drag when the element is a `button`.
```

Comment:
Should we add the link to the bug, in case somebody re-visits this in the future? https://bugzilla.mozilla.org/show_bug.cgi?id=568313

Reply:
oh wow it's actually a bug. yeah good idea